SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -226,12 +227,6 @@ void USpatialNetDriver::OnConnectedToSpatialOS()
 	}
 }
 
-void USpatialNetDriver::OnEntityPoolReady()
```

Comment:
The EntityPool dependency should not be required for BeginPlay().  It is already checked before allowing replication to start

Reply:
I think this is still needed for RPCs. If the entity pool isn't ready and we try to resolve an Actor, we'd have to queue. @improbable-valentyn to confirm.

Reply:
That's correct. RPCs can be called before the actors are replicated, but if, for example, you call an RPC with an `AActor*` argument, it needs to be resolved immediately in order to be serialized into an RPC payload. The alternative that we were using before was to queue up the non-serialized arguments, which lead to crashes when the pointers became invalid as a result of a referenced actor being destroyed before the RPC could be sent.

Reply:
Thanks for the clarification.  Would be good to have a chat about the necessary approach.  It will involve waiting for the entity id reservation response, but there are some additional conditions that require handling for that op (for example the reservation request can fail, also should we excise the op from the Worker_OpList if we're processing it early?  The Worker SDK doesn't provide a way to edit the lists from what I can see but we may be able to work around that).

Reply:
This is handled with the latest upload